infra/nightly-resources/index.js

```diff
@@ -1,46 +1,194 @@
-// Top-level load function for the main index page.
-function load() {
-  // data should be in the format of Array<{runMethod: String, benchmark: String, total_dyn_inst: Int, hyperfine: Array<{results: **hyperfine `--json` results**}>}>
-  console.log(data);
-
-  let container = document.getElementById("profile");
-  let parsed = [];
-
-  // aggregate benchmark runs into a list by their "benchmark" key
-  let groupedBy = {};
-  data.forEach((obj) => {
-    if(!groupedBy[obj.benchmark]) {
-      groupedBy[obj.benchmark] = [];
+async function getPreviousRuns() {
+    const req = await fetch("https://nightly.cs.washington.edu/reports-json/eggcc/");
+    const files = await req.json();
+    
+    // map allLinks into an objects of the shape:
+    // {branch: <git branch:string>, commit: <git commit:string>, date: <unix timestamp:int>, url: <absolute url to nightly page:string>}
+    const comparisons = [];
+    for (let i = 1; i < files.length; i++) {
+        // file name is of the format <date>:"nightly":<branch>:<commit>
+        const [date, _, branch, commit] = files[i].name.split(":");
```

Comment:
Unfortunate that we have to parse the file name but oh well

Reply:
I guess so, but can't do much when we're working with a file server, not a data-driven API.